src/util/audiosignal.h

```diff
@@ -75,8 +75,8 @@ class AudioSignal {
     AudioSignal(const AudioSignal&) = default;
```

Comment:
I didn't notice: Why have you added all those defaulted constructors and assignment operators? They are not needed. We only need to declare a virtual destructor with a default implementation.